### PR TITLE
[Enabler] [zos_fetch] Add gdg support to zos_fetch

### LINKED ISSUE #1373: [Enabler] [zos_fetch] GDG/GDS and special character support
### Is there an existing issue for this?

- [X] There are no existing issues.

### Enabler description

New MVSDataset class
-----------------------

There is a new MVSDataset class introduced in #1386 that represents a data set, when used, as part of initialization it escapes and resolves the gds relative name if any.

This is a sample on how to initialize the new class, many of these params can just be None.
```python
MVSDataSet(
            name = params.get("name"),
            record_format = params.get("record_format", None),
            volumes = params.get("volumes", None),
            data_set_type = params.get("type", None),
            block_size = params.get("block_size", None),
            record_length = params.get("record_length", None),
            space_primary = params.get("space_primary", None),
            space_secondary = params.get("space_secondary", None),
            space_type = params.get("space_type", None),
            directory_blocks = params.get("directory_blocks", None),
            key_length = params.get("key_length", None),
            key_offset = params.get("key_offset", None),
            sms_storage_class = params.get("sms_storage_class", None),
            sms_data_class = params.get("sms_data_class", None),
            sms_management_class = params.get("sms_management_class", None),
        )
```

GDG/GDS Support
-------------------

Option `src` accepts a data set name or path, enabler is to support GDS relative names.

```yaml
- name: Fetch a latest GDS in a GDG
  zos_fetch:
    src: ZOAUSER.GDG(0)
    dest: /tmp/
    flat: True

- name: Fetch a previous generation GDS in a GDG
  zos_fetch:
    src: ZOAUSER.GDG(-1)
    dest: /tmp/
    flat: True

```

Internally, in the case of a data set name being a GDS relative name we should resolve those names into absolute GDS names when the generation number is 0 or negative.

Following functions will be implemented in #1386 
```python
# Sample of soon to be implemented gdg function
if DataSet.is_gds_relative_name(name):
    src = DataSet.resolve_gds_absolute_name(src)
```

Symbols support
-----------------
 
Changes done to the BetterArgParser should be adopted in the module to not break and use the correct escaped name in case of the data set name having a symbol in it.

data set name with symbol

`ZOAUSER.TE$T`

That way, the modules will have to retrieve the new escaped named and the original user provided name inside the dictionary like:

```python
name = DataSet.escape_data_set_name(name)
```

Escaping special characters could be done via a new function inside the BetterArgParser.
These are some of the special characters:
- \#
- @
- $
- \-

For this task completion we need:
- [x] Do the development work
- [x] Update the docs to talk about GDG and GDS handling in the module.
- [x] Update the docs with at least 1 example on how to use with GDSs.
- [x] Create a new test case for GDSs.
- [x] Create a new test case for symbols.

### Ansible module

zos_fetch

### PR DESCRIPTION
**Pull request by rexemin**
##### SUMMARY
Fixes #1373.

##### ISSUE TYPE

- Enabler Pull Request

##### COMPONENT NAME
- zos_fetch
- module_util/data_set.py

### CODE REVIEW

**Comment by rexemin** [comment]
Screenshot 2024-05-27 at 11 51 59
Screenshot 2024-05-27 at 11 52 14

File: tests/functional/modules/test_zos_fetch_func.py [hunk 1]
```diff
@@ -664,3 +666,133 @@ def test_fetch_flat_create_dirs(ansible_zos_module, z_python_interpreter):
     finally:
         if os.path.exists(dest_path):
             shutil.rmtree("/tmp/" + remote_host)
+
+
+def test_fetch_sequential_data_set_with_special_chars(ansible_zos_module):
+    hosts = ansible_zos_module
+    TEST_PS = "USER.TEST.@N$IBLE#"
```
Comment (AndreMarcel99) [inline tests/functional/modules/test_zos_fetch_func.py:673]: I recommend for the name with symbols we have the option of get_tmp_ds_name(symbols=True)

**Review by AndreMarcel99 — CHANGES_REQUESTED** [review]

File: tests/functional/modules/test_zos_fetch_func.py (hunk 1 above)
Comment (rexemin) [reply tests/functional/modules/test_zos_fetch_func.py]: Done

**Review by fernandofloresg — APPROVED** [review]
Great work Ivan

**Pull request merged**